server/viame_server/viame_detection.py

```diff
@@ -45,11 +45,13 @@ def _get_clip_meta(self, folder):
 
         videoUrl = None
         # Find a video tagged with an h264 codec left by the transcoder
-        video = Item().findOne({'folderId': folder['_id'], 'meta.codec': 'h264',})
-        if video:
-            videoUrl = (
-                f'/api/v1/item/{str(video["_id"])}/download?contentDisposition=inline'
-            )
+        item = Item().findOne({'folderId': folder['_id'], 'meta.codec': 'h264',})
+        if item:
+            video = Item().childFiles(item)[0]
+            if video:
```

Comment:
Drop this condition. An item tagged with `meta.codec` should be an item with one file.  If the child file is missing, that's effectively a database schema violation and a 500 error would be appropriate.

Alternatively, keep the if and add an else that throws a more verbose exception.

Reply:
I originally didn't have it in because I assumed that it should have a single file child, I reviewed and didn't know if forever (in terms of future updates) it would have a single file.  I'll drop it.

Reply:
Yeah that's sort of the trouble with girder overall.  When you base your entire app on freeform metadata inside folders and items, your schemas get expressed throughout the whole project as code.

By putting an `if` here you say "I have a union type" and then you have to hope that's consistent with all the other code that manipulates this data.

It scales pretty poorly.  That was a large part of my motivation for writing the postprocessing endpoint: I wanted to get all that data-affirming code in one place and let that function be the source of truth for schema.  

I guess I should add some logic in that function to verify the results of the processing pipelines.  Like, after image or video conversion runs, `postprocessing` should go through and make sure all the metadata was set.